pulp_ansible/app/tasks/collections.py

```diff
@@ -81,6 +82,27 @@ def sync(remote_pk, repository_pk, mirror):
     d_version = AnsibleDeclarativeVersion(first_stage, repository, mirror=mirror)
     d_version.create()
 
+    repo_version = repository.latest_version()
+    to_deprecate = []
+    to_deprecate_query = Q()
+    for collection, deprecated in first_stage.deprecations.items():
+        namespace, name = collection.split(".")
+        to_deprecate_query |= Q(namespace=namespace, name=name)
```

Comment:
This aggregation can happen at L#482 instead `self.deprecations[f"{namespace}.{name}"] = result["deprecated"]`. It can just build and `|=` the Q objects there.